crates/bevy_pbr/src/render/pbr_functions.wgsl

```diff
@@ -257,22 +308,101 @@ fn pbr(
 
+    if diffuse_transmission > 0.0 {
+        // NOTE: We use the diffuse transmissive color, the second Lambertian lobe's calculated
+        // world position, inverted normal and view vectors, and the following simplified
+        // values for a fully diffuse transmitted light contribution approximation:
+        //
+        // perceptual_roughness = 1.0;
+        // NdotV = 1.0;
+        // F0 = vec3<f32>(0.0)
+        // occlusion = vec3<f32>(1.0)
+        transmitted_light += ambient::ambient_light(diffuse_transmissive_lobe_world_position, -in.N, -in.V, 1.0, diffuse_transmissive_color, vec3<f32>(0.0), 1.0, vec3<f32>(1.0));
+    }
+
     // Environment map light (indirect)
 #ifdef ENVIRONMENT_MAP
     let environment_light = bevy_pbr::environment_map::environment_map_light(perceptual_roughness, roughness, diffuse_color, NdotV, f_ab, in.N, R, F0);
     indirect_light += (environment_light.diffuse * occlusion) + environment_light.specular;
+
+    // we'll use the specular component of the transmitted environment
+    // light in the call to `transmissive_light()` below
+    var transmitted_environment_light_specular = vec3<f32>(0.0);
+
+    if diffuse_transmission > 0.0 || transmission > 0.0 {
+        // NOTE: We use the diffuse transmissive color, inverted normal and view vectors,
+        // and the following simplified values for the transmitted environment light contribution
+        // approximation:
+        //
+        // NdotV = 1.0;
+        // f_ab = vec2<f32>(0.1)
+        // R = T // see definition below
+        // F0 = vec3<f32>(0.16)
+        // occlusion = 1.0
+
+        let T = -normalize(
+            in.V + // start with view vector at entry point
+            refract(in.V, -in.N, 1.0 / ior) * thickness // add refracted vector scaled by thickness, towards exit point
+        ); // normalize to find exit point view vector
+
+        let transmitted_environment_light = bevy_pbr::environment_map::environment_map_light(perceptual_roughness, roughness, diffuse_transmissive_color, 1.0, vec2<f32>(0.1), -in.N, T, vec3<f32>(0.16));
+        transmitted_light += transmitted_environment_light.diffuse;
+        transmitted_environment_light_specular = transmitted_environment_light.specular;
+    }
+#else
+    // If there's no environment map light, there's no transmitted environment
+    // light specular component, so we can just hardcode it to zero.
+    let transmitted_environment_light_specular = vec3<f32>(0.0);
 #endif
 
     let emissive_light = emissive.rgb * output_color.a;
 
+    if transmission > 0.0 {
+        transmitted_light += lighting::transmissive_light(in.world_position, in.frag_coord.xyz, in.N, in.V, ior, thickness, perceptual_roughness, transmissive_color, transmitted_environment_light_specular).rgb;
+    }
+
+    if in.material.attenuation_distance < (1.0 / 0.0) /* f32::INFINITY */ {
+        // We reuse the `atmospheric_fog()` function here, as it's fundamentally
```

Comment:
Maybe we rename the fog function to volumetric_scattering or something?

Reply:
It's only a _very rough_ approximation of that, I feel like that name might imply something more proper?

Reply:
approximate_volumetric_scattering()? Idk

Reply:
Hmm.. Thought about it more, I think I'd like to keep the function name consistent with the other fog types (`linear_fog`, `exponential_fog`, `exponential_squared_fog`) primarily for the sake of how they're organized/located in the codebase. 

Also, might be a matter of personal preference, but I feel like the more mundane name (`atmospheric_fog`) makes it easier to understand what's going on by analogy/intuition, rather than a more technical name about the internal behavior.